src/lib/util/state.ts

```diff
@@ -132,6 +136,10 @@ export const updateCode = (
 	}
 
 	inputStateStore.update((state) => {
+		if (resetPanZoom) {
```

Comment:
Should we store the state on the pan/zoom? It makes sense on larger diagrams but when making a diagram from the start, it can require the user to hit reset often. Maybe we can store the pan and zoom state only if the user has auto-sync turned off or their diagram is larger than some limit?

Reply:
Nice catch there @Yash-Singh1!
I've added a pan&zoom toggle, which should resolve the issue and give the user more control.